cpp/src/io/statistics/column_stats.cu

```diff
@@ -295,73 +213,54 @@ void __device__ gatherStringColumnStats(stats_state_s *s, uint32_t t, Storage &s
 {
   using block_reduce = cub::BlockReduce<uint32_t, block_size>;
   uint32_t len_sum   = 0;
-  const char *smin   = nullptr;
-  const char *smax   = nullptr;
-  uint32_t lmin      = 0;
-  uint32_t lmax      = 0;
   uint32_t nn_cnt    = 0;
-  bool has_minmax;
-  string_stats minval, maxval;
+  bool has_minmax    = false;
+
+  string_view minimum_value = string_view::max();
+  string_view maximum_value = string_view::min();
 
   for (uint32_t i = 0; i < s->group.num_rows; i += block_size) {
-    uint32_t r                = i + t;
-    uint32_t row              = r + s->group.start_row;
-    const uint32_t *valid_map = s->col.valid_map_base;
-    uint32_t is_valid         = (r < s->group.num_rows && row < s->col.num_values)
-                          ? (valid_map) ? (valid_map[(row + s->col.column_offset) >> 5] >>
-                                           ((row + s->col.column_offset) & 0x1f)) &
-                                            1
-                                        : 1
+    uint32_t r        = i + t;
+    uint32_t row      = r + s->group.start_row;
+    uint32_t is_valid = (r < s->group.num_rows && row < s->col.leaf_column->size())
+                          ? s->col.leaf_column->is_valid(row)
```

Comment:
This is incorrect for list columns because in case of sliced tables, the leaf column does not account for the offset.

Reply:
Bringing the discussion here:
We may be able to remove the check altogether. The check for num_rows should be enough. num_rows is already set to num_leaf_values so it cannot exceed the column size anyway. I cannot think of the case where `r < s->group.num_rows` wouldn’t automatically mean `row < s->col.leaf_column->size()`